### PR TITLE
libfreenect2 0.1.1 (new formula)

### PR DESCRIPTION
**Pull request by RyanGordon**
This adds the libfreenect2 formula for first stable version v0.1.1.

### CODE REVIEW

File: Library/Formula/libfreenect2.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  head "https://github.com/OpenKinect/libfreenect2.git"
+
+  option :universal
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libusb"
+  depends_on "jpeg-turbo"
+  depends_on "homebrew/versions/glfw3"
```
Comment (rakkesh) [inline Library/Formula/libfreenect2.rb]: A formula in `Homebrew` core can not have `depends_on` for a formula in a `Tap`, unless, it is an `:optional` dependency.

**Comment by rakkesh** [comment]
👍🏽

Could you add `glfw` into `Homebrew` core?

File: Library/Formula/libfreenect2.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/libfreenect2.rb]: ^ :+1:

**Comment by RyanGordon** [comment]
@rakkesh Just so I understand correctly, do you want me to create a separate PR for a new glfw formula for this repo? Any specific version I should target?

File: Library/Formula/libfreenect2.rb (hunk 1 above)
Comment (RyanGordon) [inline Library/Formula/libfreenect2.rb]: Ah thanks, I will look into that

**Comment by rakkesh** [comment]
You need to add the formula, `glfw` in this PR. You can take up the version which is needed by `libfreenect`; I am guessing version `3.1.2` can be used.

**Comment by RyanGordon** [comment]
Hi @rakkesh,

I have added the additional glfw module in https://github.com/RyanGordon/homebrew/commit/934b23a7ed689e2d868e9d886235bb0cd84356af, taken directly from https://github.com/Homebrew/homebrew-versions/blob/master/glfw3.rb. Currently 3.1.2 is the latest released version available for GLFW and seems to work just fine for libfreenect2

**Comment by rakkesh** [comment]
@RyanGordon, thanks. There are a few audit errors in `libfreenect2` formula, and also, `glfw` formula needs to be committed in git prior to `libfreenect2`.

```
brew audit libfreenect2 --strict --online
Stacktrace

        ==> brew style libfreenect2
== /usr/local/Library/Formula/libfreenect2.rb ==
C: 20: 33: %r-literals should be delimited by { and }.
C: 20: 33: Use // around regular expression.
C: 21: 33: %r-literals should be delimited by { and }.
C: 21: 33: Use // around regular expression.

1 file inspected, 4 offences detected
```

**Comment by RyanGordon** [comment]
Hi @rakkesh,
I've swapped the order of the commits and fixed the audit errors. Anything else I should do?

Thanks!

**Comment by rakkesh** [comment]
There are a audit problems in `glfw` formula which need to be fixed,
```
$ brew audit --strict --online glfw
==> audit problems
glfw:
 * File should end with a newline
 * Use '--with-test' instead of '--with-tests'. Migrate '--with-tests' with `deprecated_option`.
 
Error: 2 problems in 1 formula
```
Also, remove the patch in `glfw`, which changes the library link name to `-lglfw3`, which causes build failure for `libfreenect2`.

**Comment by RyanGordon** [comment]
Hi @rakkesh,
Ah I see, I fixed the audit problems and removed the patch. Libfreenect2 formula seems to install on my machine as well still

**Comment by DomT4** [comment]
Need to talk about the `glfw` situation, but will have to be after I've taken something to get rid of this headache :smile:.

**Comment by mikemcquaid** [comment]
@DomT4 not sure if you want this merged or not so will leave it to you.

**Comment by DomT4** [comment]
No, we'll need to address the `glfw` problem. At the moment the `glfw` is a direct port of the one in Versions, which is fine in a way because at this point it probably could live in the core. This was raised before but not followed through with: #43563.

However, if we're going to move it the formula needs to retain the `glfw3` name so people who currently have it installed don't lose the installation when it gets removed from Versions and we don't have to deal with linking conflicts all over the place.

On top of that we need to keep it the libs with `3` appended, as in `glfw3` for compatibility. We could do this via symlinks, so you get versioned and unversioned ones installed.

**Comment by apjanke** [comment]
Closing this due to inactivity. If you'd still like to pursue this, feel free to respond to the previous comments/questions and re-open on the new [`homebrew/core` repo](https://github.com/homebrew/homebrew-core).

**Pull request closed without merging**